### ISSUE TITLE
Susccessfully downlaoded audio files no longer show up when app is offline.

### ISSUE THREAD

**Issue by samtrek** [GrakovNe/lissen-android#90]
Successfully downloaded audio files do not show up when app is offline, cannot state from which version this started
Lissen v 1.1.25
Android 13

**Comment by GrakovNe** [comment]
could i ask you reinstall app and re-download books/podcasts?

Few versions ago the downloading mechanism has been changed and it may be truth that backward compatibility doesn't working properly in your case. If so, re-install and re-download shall fix the problem

I've checked it on the latest version (1.1.25) and can say that cached files are playable, but i didn't checked if it works between previous versions

**Comment by samtrek** [comment]
I did as you advised, the first downloaded audiobook showed up, but the second did not show up.

**Comment by GrakovNe** [comment]
have to research it deeply. Could you share for a moment your abs server with me? If so, create please account for me and send credentials securely to grakovne@gmail.com

**Comment by samtrek** [comment]
My own abs is on a  local network, I only achieve some sort of networking through tailscale, if you are using tailscale maybe it will be possible.

**Comment by GrakovNe** [comment]
i'l use everything what needs to connect your server. Even Tailscale) Just send me server address and TS exit node address

**Comment by GrakovNe** [comment]
i did accept your TS invite, but i still need ABS host address and credentials FYI

**Comment by samtrek** [comment]
Sent

Sent from Proton Mail Android

-------- Original Message --------

**Comment by GrakovNe** [comment]
got it, thank you much. After a weekend i'll try to research this issue

**Comment by GrakovNe** [comment]
got connected via your ip and 13378 port, thanks. Could you point me to book that can't be downloaded properly?

**Comment by samtrek** [comment]
listening to a series `mark of the fools`, a large file it may take a while to download, you may download a couple of books to test it out. I will suggest Erich Fromm books for testing the size are not that large.

**Comment by GrakovNe** [comment]
seems like i got reproduced this issue. It's application bug caused by the connectivity issues

Could i ask to start download something and don't leave the player screen before download finished. If my suggestion is right - everything will works in this case

**Comment by samtrek** [comment]
ok seems if I try to download a currently playing media, it becomes successful, but I prefer to download audiobooks I will like to listen to ahead while playing another audiobook, can this be fixed?

**Comment by GrakovNe** [comment]
yes, it **shall** be fixed soon, it's kinda major bug and thank you for the report. I just asked you to do this because a wanted to define the root problem

keep your eyes on this ticket, i'll track the progress

**Comment by samtrek** [comment]
ok I will give my feedback once it has been fixed, thanks for the prompt response.

**Comment by GrakovNe** [comment]
https://github.com/GrakovNe/lissen-andrIn [release 1.3.5](https://github.com/GrakovNe/lissen-android/releases/tag/1.3.5), the download mechanism has been completely reworked. You can now download books and podcasts even when the app is in the background or your screen is turned off. The issue that prevented multiple books or podcasts from being downloaded in parallel has also been fixed.oid/releases/tag/1.3.5

**Comment by samtrek** [comment]
thank you for the update, but it seems something is wrong, the downloader shows `content is caching`, I have reinstalled the app to be sure, and also their is no percentage to show how much of the audiobook has been download, a new downloader design?

**Comment by GrakovNe** [comment]
Yes, the proper progress bar will be brought back soon — it's just a current limitation of the new content downloading approach.

**Comment by GrakovNe** [comment]
@samtrek by the way, the release https://github.com/GrakovNe/lissen-android/releases/tag/1.3.7 **have** proper progress bar

![Image](https://github.com/user-attachments/assets/7ebc7f82-0d04-44d5-9b8f-9c7fb8688088)

**Comment by samtrek** [comment]
Yes the new version has the progress bar, if I will not be too greedy can we have the estimated time to finish download back, if not possible totally fine by me.
Thank you for the good works
